src/clojure/puppetlabs/ssl_utils/simple.clj

```diff
@@ -75,27 +96,41 @@
       (:not-after validity)
       (:x500-name host-keys)
       (:public-key host-keys)
-      [])))
+      extensions))))
 
 (defn gen-cert
-  [certname ca-cert serial]
-  {:pre [(string? certname)
-         (ssl-cert? ca-cert)
-         (integer? serial)]
-   :post [(ssl-cert? %)]}
-  (let [cert-keys (gen-keys certname)]
-    (assoc cert-keys :cert (gen-cert*
-                             (dissoc ca-cert :cert)
-                             cert-keys
-                             serial))))
+  "Generate a certificate. An optional map may be provided to specify:
+
+   * :keylength   Bit length to use for the public/private keys;
+                  defaults to 4096.
+   * :extensions  List of certificate extensions to include on the certificate;
+                  defaults to []."
+  ([certname ca-cert serial] (gen-cert certname ca-cert serial {}))
+  ([certname ca-cert serial options]
+   {:pre [(string? certname)
+          (ssl-cert? ca-cert)
+          (integer? serial)
+          (map? options)]
+    :post [(ssl-cert? %)]}
+   (let [cert-keys (gen-keys certname options)]
+     (assoc cert-keys :cert (gen-cert* ca-cert cert-keys serial options)))))
```

Comment:
You referring to the previous `dissoc`? No, that was a leftover from when this code was guarded by Schema, which it no longer is. See #56 for more information